### PR TITLE
Fixed issue with missing alarms

### LINKED ISSUE #9706: Health stopped working
##### Bug report summary
While I was testing the PR #9698 I observed that initially I did not receive any alarm. After this I decided to clean up the `dbengine` files and starting my tests again, it worked during few hours, but after I change from a branch to another and recompile I stopped to receive any alarm.

In the first time that the problem happend, I observed that I had thousands of errors like this:

```
Ignoring metric with out of order timestamp
``` 

but, when the problem happens again, I did not receive more this error message.

##### OS / Environment
```
Linux hades 5.4.57 #1 SMP Sat Aug 8 12:10:00 UTC 2020 x86_64 Intel(R) Core(TM) i7-7500U CPU @ 2.70GHz GenuineIntel GNU/Linux
/etc/os-release:NAME=Slackware
/etc/os-release:VERSION="14.2"
/etc/os-release:ID=slackware
/etc/os-release:VERSION_ID=14.2
/etc/os-release:PRETTY_NAME="Slackware 14.2 x86_64 (post 14.2 -current)"
/etc/os-release:ANSI_COLOR="0;34"
/etc/os-release:CPE_NAME="cpe:/o:slackware:slackware_linux:14.2"
/etc/os-release:HOME_URL="http://slackware.com/"
/etc/os-release:SUPPORT_URL="http://www.linuxquestions.org/questions/slackware-14/"
/etc/os-release:BUG_REPORT_URL="http://www.linuxquestions.org/questions/slackware-14/"
/etc/os-release:VERSION_CODENAME=current
```

##### Netdata version
netdata v1.24.0-8-ga15e450e2
 

##### Component Name
Health
Dbengine

##### Steps To Reproduce
1. Start Netdata
2. Configure an alarm, for example:

```
 alarm: dev_dim_template
    on: system.cpu
    os: linux
lookup: sum -3s at 0 every 3 percentage foreach *
 units: %
 every: 10s
  warn: $this > 1
  crit: $this > 4
```
3. Stop Netdata ( I did not test sending -USR2, all the problems happening when I was stopping Netdata, recompiling and starting it again)
4. Verify alarms and logs.

##### Expected behavior
To have alarms running, independent of the Netdata version or branch, on my case, I used the cited PR and the master branch, and I had problems with both.

### PR DESCRIPTION
**Pull request by stelfrag**
Fixes #9706
##### Summary
When the health subsystem is configured, archived charts are skipped to avoid unnecessary configuration that slows
down the metadata log replay. 

Health related configuration for those charts is performed when a chart switches from archived to active by having a collector gather metrics for any dimension of that chart.

The following init code (found in the health configuration initialization) did not run in that case

```
rrddim_foreach_read(rd, st) {
    rrdcalc_link_to_rrddim(rd, st, host);
}
```

##### Component Name
database
health

##### Test Plan
- Before the PR
- Configure an alarm as described in #9706 
- Restart the agent 
   - The alarm is not raised
- Perform health reload
- Alarm is now raised
  - Note that during health reload the init process is done properly because the chart is not archived anymore

- Apply the PR    
- Restart agent
- Alarm is now immediately raised

### CODE REVIEW

**Comment by stelfrag** [comment]
@mfundul you are correct. Needs to move to dim creation

**Comment by mfundul** [comment]
@stelfrag you should run coverity on your branch since you moved flagged code.

File: database/rrddim.c [hunk 1]
```diff
@@ -3,6 +3,32 @@
...
+        for (count = 0; count < 5; count++) {
+            hostlocked = netdata_rwlock_trywrlock(&host->rrdhost_rwlock);
+            if (!hostlocked) {
+                rrdcalc_link_to_rrddim(rd, st, host);
+                rrdhost_unlock(host);
+                break;
+            } else if (hostlocked != EBUSY) {
+                error("Cannot lock host to create an alarm for the dimension.");
+            }
+            usleep(200000);
```
Comment (thiagoftsm) [inline database/rrddim.c:22]: I will give a suggestion here, instead to use `usleep`, please, change to `sleep_usec` that is declared inside `libnetdata/clocks/clocks.c`.

**Comment by stelfrag** [comment]
> @stelfrag you should run coverity on your branch since you moved flagged code.

yes it produces a warning about deadlock in that part of code (currently marked as false positive in netdata/netdata)

File: database/rrddim.c (hunk 1 above)
Comment (stelfrag) [reply database/rrddim.c]: Thanks @thiagoftsm -- I changed it

**Comment by amoss** [comment]
> > @stelfrag you should run coverity on your branch since you moved flagged code.
> 
> yes it produces a warning about deadlock in that part of code (currently marked as false positive in netdata/netdata)

If the old flag had an owner then get them to recheck the moved code, otherwise it predates people owning flags - in which case: congratulations! Please verify it does not deadlock (with two confirmations) and put your name in the field.

**Pull request merged**